pandas/tests/frame/test_analytics.py

```diff
@@ -1965,3 +1899,111 @@ def test_dot(self):
 
         with tm.assertRaisesRegexp(ValueError, 'aligned'):
             df.dot(df2)
+
+
+@pytest.fixture
+def df_duplicates():
+    return pd.DataFrame({'a': [1, 2, 3, 4, 4],
+                         'b': [1, 1, 1, 1, 1],
+                         'c': [0, 1, 2, 5, 4]},
+                        index=[0, 0, 1, 1, 1])
+
+
+@pytest.fixture
+def df_strings():
+    return pd.DataFrame({'a': np.random.permutation(10),
+                         'b': list(ascii_lowercase[:10]),
+                         'c': np.random.permutation(10).astype('float64')})
+
+
+class TestNLargestNSmallest(object):
+
+    # ----------------------------------------------------------------------
+    # Top / bottom
+    @pytest.mark.parametrize(
+        'n, order',
+        product(range(1, 11),
+                [['a'],
+                 ['c'],
+                 ['a', 'b'],
+                 ['a', 'c'],
+                 ['b', 'a'],
+                 ['b', 'c'],
+                 ['a', 'b', 'c'],
+                 ['c', 'a', 'b'],
+                 ['c', 'b', 'a'],
+                 ['b', 'c', 'a'],
+                 ['b', 'a', 'c'],
+
+                 # dups!
+                 ['b', 'c', 'c'],
+
+                 ]))
+    def test_n(self, df_strings, n, order):
+        # GH10393
+        df = df_strings
+
+        error_msg = (
+            "'b' has dtype: object, cannot use method 'nsmallest' "
+            "with this dtype"
+        )
+        if 'b' in order:
+            with pytest.raises(TypeError) as exception:
+                df.nsmallest(n, order)
+            assert exception.value, error_msg
+        else:
+            result = df.nsmallest(n, order)
+            expected = df.sort_values(order).head(n)
+            tm.assert_frame_equal(result, expected)
+
+        if 'b' in order:
+            with pytest.raises(TypeError) as exception:
+                df.nsmallest(n, order)
+            assert exception.value, error_msg
+        else:
+            result = df.nlargest(n, order)
+            expected = df.sort_values(order, ascending=False).head(n)
+            tm.assert_frame_equal(result, expected)
+
+    def test_n_error(self, df_strings):
```

Comment:
prob should expand this with a different df that has pretty much all dtypes, then can test ones which have an error

heres a sample
```
        df = pd.DataFrame(
            {'group': [1, 1, 2],
             'int': [1, 2, 3],
             'float': [4., 5., 6.],
             'string': list('abc'),
             'category_string': pd.Series(list('abc')).astype('category'),
             'category_int': [7, 8, 9],
             'datetime': pd.date_range('20130101', periods=3),
             'datetimetz': pd.date_range('20130101',
                                         periods=3,
                                         tz='US/Eastern'),
             'timedelta': pd.timedelta_range('1 s', periods=3, freq='s')},
            columns=['group', 'int', 'float', 'string',
                     'category_string', 'category_int',
                     'datetime', 'datetimetz',
                     'timedelta'])
```